api/client/images.go

```diff
@@ -18,26 +19,26 @@ import (
 )
 
 // FIXME: --viz and --tree are deprecated. Remove them in a future version.
-func (cli *DockerCli) WalkTree(noTrunc bool, images *engine.Table, byParent map[string]*engine.Table, prefix string, printNode func(cli *DockerCli, noTrunc bool, image *engine.Env, prefix string)) {
-	length := images.Len()
+func (cli *DockerCli) WalkTree(noTrunc bool, images *[]*types.Image, byParent map[string]*[]*types.Image, prefix string, printNode func(cli *DockerCli, noTrunc bool, image *types.Image, prefix string)) {
```

Comment:
Seems like `*[]*` type here is overkill. As I see those slices not changed during `WalkTree`.